src/adapter/src/coord.rs

```diff
@@ -2573,6 +2568,23 @@ impl Coordinator {
     }
 }
 
+/// Contains information about the last message the [`Coordinator`] processed.
+#[derive(Debug)]
+struct LastMessage {
+    kind: &'static str,
+    stmt: Option<Arc<Statement<Raw>>>,
+}
+
+impl Drop for LastMessage {
+    fn drop(&mut self) {
+        // Only print the last message if we're currently panicking, otherwise we'd spam our logs.
+        if std::thread::panicking() {
+            // If we're panicking theres no guarantee `tracing` still works, so print to stderr.
+            eprintln!("Coordinator panicking, dumping last message\n{:?}", self);
```

Comment:
using `tracing::error!` here would be preferred but I don't think we're guaranteed to have tracing work. But I'm also not sure how easy it is to read stderr?

Reply:
We should do the same thing we do for logging in the stuck coordinator case and redact the SQL, to keep customer data out of logs.

Reply:
But yeah, agreed on `eprintln!` rather than `tracing::error!`. I think it should go to Loki just fine.

Reply:
Good call! Updated to use the redacted SQL